pkg/operator/hive/hive_controller.go

```diff
@@ -298,10 +300,15 @@ func (r *ReconcileHiveConfig) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	// Initialize hive ready condition if not present to Unknown
+	instance.Status.Conditions = util.SetHiveConfigCondition(instance.Status.Conditions, hivev1.HiveReadyCondition, corev1.ConditionUnknown, hivev1.InitializedConditionReason, "Condition Initialized")
```

Comment:
Technically you're setting this condition on every reconcile before it eventually gets set to True or False further in the code - so particularly for this scenario it would be better if you try to find the condition first before calling this line of code